### PR TITLE
Fixed bug: UI component can't click correctly by moving UI and camera far away of origin.

### PR DESCRIPTION
**Pull request by yangws**
Related PR: https://github.com/cocos2d/cocos2d-x/pull/12419

### CODE REVIEW

**Comment by yangws** [comment]
[ci rebuild]

**Comment by yangws** [comment]
[ci rebuild]

**Comment by yangws** [comment]
@pandamicro @dabingnn

**Comment by pandamicro** [comment]
@dabingnn Please review this PR

File: cocos/ui/UIWidget.h [hunk 1]
```diff
@@ -956,6 +958,7 @@ class CC_GUI_DLL Widget : public ProtectedNode, public LayoutParameterProtocol
     Vec2 _positionPercent;
 
     bool _hitted;
+    const Camera *_hittedByCamera;
```
Comment (dabingnn) [inline cocos/ui/UIWidget.h:963]: What is the purpose of this param? Is it used to record gestures such as drag?

Comment (yangws) [inline cocos/ui/UIWidget.h:963]: // weak reference of the camera which made the widget passed the hit test when response touch begin event
    // it's useful in the next touch move/end events
    const Camera *_hittedByCamera;

**Comment by dabingnn** [comment]
@pandamicro This pr could be merged. Thanks.

**Pull request merged**